spirv_msl.hpp

```diff
@@ -319,6 +326,7 @@ class CompilerMSL : public CompilerGLSL
 		uint32_t shader_input_buffer_index = 22;
 		uint32_t shader_index_buffer_index = 21;
 		uint32_t shader_patch_input_buffer_index = 20;
+		uint32_t draw_info_index = 20;
```

Comment:
Conflict with shader_patch_input_buffer_index?

Reply:
Yes, though tessellation is not yet supported together with geometry shaders. 19 felt a bit low, but I guess it can default to that.

Reply:
> Yes, though tessellation is not yet supported together with geometry shaders

what is the plan to implement this?

Reply:
There's no concrete plan, partly because, as far as I know, we have not seen a game that needs it. I'm guessing that we'll want to bake tessellation programs into the object stage and generate a tessellated payload (i.e. multiple primitives instead of just one), and then spawn a larger mesh thread grid over it. But yes, it needs to be hashed out.

Reply:
Seeing as MoltenVK advertises tessellation, that needs to be solved *before* geometry shaders  could be merged…

Reply:
That's a fine point to raise in the corresponding MoltenVK PR, and I don't completely disagree, but I see no reason to delay upstreaming of useful, if incomplete, functionality. It should be fine spec-wise if MoltenVK decides to just not advertise geometry shaders until it makes sense to, and on our side that will lift the burden of rebasing until it's done.

Reply:
Just gonna leave this here:

![Screenshot_2023-06-11-04-58-43-84_40deb401b9ffe8e1df2f1cc5ba480b12.jpg](https://github.com/KhronosGroup/SPIRV-Cross/assets/1131720/2d7ad760-1d13-41f6-b010-15e9e836413b)